doc/docs/master/deploy-manage/deploy/deploy-pachyderm-jupyterhub.md

```diff
@@ -0,0 +1,245 @@
+   STATUS: deployed
+   REVISION: 1
+   TEST SUITE: None
+   NOTES:
+   Thank you for installing JupyterHub!
+   ...
+   ===> notes
+   - Since Pachyderm auth doesn't appear to be enabled, JupyterHub will expect the following global password for users: XXXXXXXXXXXXXXXXXXXXX
+   ```
+
+   The JupyterHub is now deployed on your Kubernetes cluster.
+
+## Log in to JupyterHub
+
+After you deploy JupyterHub, you can access the JupyterHub UI
+in a web browser through the Pachyderm cluster hostname on port
+`80`.
+
+If you have enabled Pachyderm access controls, use your Pachyderm
+token to log in. Otherwise, use the global password
+that was printed out when you ran the `./init.py` script with any
+username.
+
+## Using `python-pachyderm`
+
+After you log in, you can use the [Python Pachyderm](https://pachyderm.github.io/python-pachyderm/python_pachyderm.m.html#header-functions)
+client API to manage Pachyderm directly in Jupyter notebooks.
+
+The following lines initialize the Python Pachyderm client in JupyterHub:
+
+```bash
+import python_pachyderm
+client = python_pachyderm.Client.new_in_cluster()
+```
+
+!!! note
+    This syntax is different from the one you run directly
+    from your terminal.
+
+Then, you can API request to create Pachyderm repositories, pipelines,
+put files, and others. For example, you can check the current user by
+running the following:
+
+```bash
+import python_pachyderm
+client = python_pachyderm.Client.new_in_cluster()
+print(client.who_am_i())
+```
+
+The following screenshot demonstrates how this looks in JupyterHub:
+
+![JupyterHub whoami](../../assets/images/s_jupyterhub_whoami.png)
+
+To create a repository in Pachyderm, complete the following
+steps:
+
+1. In the JupyterHub UI, create a new notebook by clicking **New**
+and selecting **Python 3**.
+1. In the dialog field, type:
+
+   ```bash
+   import python_pachyderm
+   client = python_pachyderm.Client.new_in_cluster()
+   client.create_repo('test')
+   client.list_repo()
+   ```
+
+1. Click **Run**.
+
+   **System Response:**
+
+   ```bash
+   [repo {
+   name: "test"
+   }
+   created {
+     seconds: 1576869000
+     nanos: 886123695
+   }
+   auth_info {
+     access_level: OWNER
+   }
+   ]
+   ```
+
+1. Delete the repository:
+
+   ```bash
+   import python_pachyderm
+   client = python_pachyderm.Client.new_in_cluster()
+   client.delete_repo('test')
+   client.list_repo()
+   ```
+
+   **System Response:**
+
+   ```bash
+   []
+   ```
+
+!!! note "See also:"
+    [Image Detection on JupyterHub Example](https://github.com/pachyderm/jupyterhub-pachyderm/blob/master/doc/opencv.md)
+
+## Update Your Pipeline
+
+When you need to update your pipeline,
+you can do so directly in the JupyterHub UI by
+modifying the corresponding notebook and
+reruning it again.
+
+If you need to update your code, you can do it in the
+JupyterHub UI as well. The main advantage of
+using Python Pachyderm Python client is that you
+do need to rebuild your pipeline image to apply
```

Comment:
do or don't? need to rebuild pipeline images?

Reply:
this is a potentially fancy feature. We need to describe it in more detail. Do you need to use reprocess? if so, how? Is a new image being built behind the scenes? what's actually going on here and how much "smartness" is there?

These details can be covered in TOH with Yusuf and then we can document them more

Reply:
yes, "do not" of course. Here is more details: https://pachyderm.github.io/python-pachyderm/python_pachyderm.m.html

Reply:
@JoeyZwicker this is a utility function exposed by python-pachyderm. I'm not sure how much detail we should go into here. On the one hand, it seems useful for JupyterHub users because they can deploy code much more quickly to pachyderm, and semantically it's quite different from the typical `pachctl create pipeline`. On the other hand, there's a good amount of detail to it, as you can see in the library docs.

To answer your questions:
1) Similar to `pachctl create pipeline`, you can choose to reprocess or not.
2) It doesn't built an image behind the scenes. Instead it creates a PFS repo with the python source code, and a pipeline (with a standard python image) for downloading/building the dependencies. The new pipeline then depends on these repos, as well as the input repo(s) specified by the user.